Improve filters documentation (#115)
ambanum/CGUs: CONTRIBUTING.md

These filters are declared as JavaScript functions that modify the downloaded web page through the [DOM API](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model).

Filters take the document and the document declaration as parameters and are:

- **in-place**: they modify the document structure and content directly;
- **idempotent**: they should return the same document structure and content even if run repeatedly on their own result.

Filters are loaded automatically from files named after the service they operate on. For example, filters for the Meetup service, which is declared in `services/Meetup.json`, are loaded from `services/Meetup.filters.js`.

The generic function signature for a filter is:

```js
export [async] function filterName(document [, documentDeclaration])
```

Each filter is exposed as a named function export that takes a `document` parameter and behaves like the `document` object in a browser DOM. These functions can be `async`, but they will still run sequentially.

> > It is actually a [JSDOM](https://github.com/jsdom/jsdom) document instance.

Reviewer: ```suggestion
> The `document` parameter is actually a [JSDOM](https://github.com/jsdom/jsdom) document instance.
```

> There is also an optional parameter containing the whole document declaration. It can be useful if you need to access the defined document url or selector inside your filter.

Reviewer: > All parameters are “optional” in the sense that they are provided to the function, you may or may not use them.

```suggestion
The second parameter contains the whole document declaration. It can be useful if you need to access the defined document URL or selector inside your filter.
```

Filters are loaded automatically from files named after the service they operate on. For example, filters for the Meetup service, which is declared in `services/Meetup.json`, are loaded from `services/Meetup.filters.js`.

> Each filter is exposed as a named function export, which could be `async`, that takes a `document` parameter and behaves like the `document` object in a browser DOM.

Reviewer: If I understand correctly, I feel this is a useful precision:
```suggestion
Each filter is exposed as a named function export that takes a `document` parameter and behaves like the `document` object in a browser DOM. These functions can be `async`, but they will still run sequentially.
```

> It is actually a [JSDOM](https://github.com/jsdom/jsdom) document instance.

There is also an optional parameter containing the whole document declaration. It can be useful if you need to access the defined document url or selector inside your filter.

> For example, you can scope your the selection in your filter by the `selector` defined in the document declaration:

Reviewer: ```suggestion
For example, you can scope the selection in your filter with the `selector` defined in the document declaration:
```

```js
export function removeImages(document, { select: selector }) {
  const images = document.querySelectorAll(`${selector} img`);

>   images.forEach(el => el.remove());

Reviewer: > This documentation is very important as it sets the standard for what we expect 🙂
> Do we want to encourage this or `document.querySelectorAll(`${selector} img`).forEach(el => el.remove());`?
> 
> It's a real question, I'm not sure I have an opinion on this.

Author: I do not have a strong opinion on this neither, so I suppose it is no so important :).

}

```

So the generic function signature is:

```js
export [async] function filterName(document [, documentDeclaration])

> ```

Reviewer: Maybe move this to the top of the description, right before “each filter is exposed as a named function export”? This way, if the contributor is experimented she doesn't feel obligated to read the entire following paragraphs  🙂

> Learn more about [usual noise](https://github.com/ambanum/CGUs/wiki/Usual-noise).

Reviewer: This feels out of place now, maybe move it up or after the examples paragraph?

> You can find example of filters in [`/services/*.filters.json`](./services) files.

Reviewer: ```suggestion
You can find examples of filters in [`/services/*.filters.json`](./services) files.
```
